### PR TITLE
Update logger to output to the correct path - Closes #5329

### LINKED ISSUE #5329: Logs are not saved under the rootPath
### Expected behavior
Logs should be saved under the `<rootPath>/<label>/logs`,

### Actual behavior
logs is saved in different place than the above folder

### Steps to reproduce
Start test_app with default setting and check `~/.lisk/` folder

### Which version(s) does this affect? (Environment, OS, etc...)
development

### Additional Information
- Logger should be converted to typescript
- Logger should be moved to `application` and update the config

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #5329 

### How was it solved?

- Update to use `dirs` defined in the application
- Move logger from `components` to `application`

### How was it tested?

- All the tests are updated
- Check logs are written in the `rootPath/label` when using the `test_app`

### CODE REVIEW

File: framework/src/modules/http_api/http_api.js [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 'use strict';
 
+// eslint-disable-next-line
 const { createLoggerComponent } = require('../../components/logger');
```
Comment (nazarhussain) [inline framework/src/modules/http_api/http_api.js:18]: I think this logger is removed.

File: framework/src/application/application.ts [hunk 2]
```diff
@@ -109,6 +109,11 @@ export interface ApplicationConfig {
 		readonly defaultPassword?: string;
 	};
 	readonly network: NetworkConfig;
+	readonly logger: {
```
Comment (ManuGowda) [inline framework/src/application/application.ts:112]: export `LoggerInput` and use `Partial< LoggerInput>` ?

File: framework/src/application/application.ts [hunk 3]
```diff
@@ -338,10 +339,15 @@ export class Application {
...
-		await this._forgerDB.close();
-		await this._networkDB.close();
+		try {
+			await this._network.cleanup();
+			await this._node.cleanup();
+			await this._blockchainDB.close();
+			await this._forgerDB.close();
+			await this._networkDB.close();
+		} catch (error) {
+			this.logger.fatal({ err: error as Error }, 'failed to shutdown');
```
Comment (ManuGowda) [inline framework/src/application/application.ts:349]:
```suggestion
			this.logger.fatal({ err: error as Error }, 'Failed to shutdown');
```

File: framework/src/application/application.ts (hunk 2 above)
Comment (ManuGowda) [reply framework/src/application/application.ts:112]: ?

**Pull request merged**